torch/csrc/CudaIPCTypes.cpp

```diff
@@ -0,0 +1,156 @@
+#ifdef USE_CUDA
+#include <torch/csrc/CudaIPCTypes.h>
+#include <map>
+#include <mutex>
+
+namespace torch {
+
+void BadIPCPatternError() {
```

Comment:
This is a private function, so the name doesn't matter too much, but I do find the current name very un-enlightening. Perhaps `warnProducerTerminatedBeforeSharedTensorsReleased`?